apis/r/R/utils.R

```diff
@@ -71,7 +71,7 @@ arrow_to_dt <- function(arrlst) {
 }
 
 ##' @noRd
-as_arrow_table <- function(arrlst) {
+to_arrow_table <- function(arrlst) {
```

Comment:
I could possibly make that simple list of two external pointers a simple S3 class (which I considered for the simple features like pretty printing).  That may open the door for a dispatch of as_arrow_table.CLASSHERE as in your coercion utilities.  Is that better?

In the short term the renaming is fine but I do like these "verbs" to start with 'as' ...

Reply:
> I could possibly make that simple list of two external pointers a simple S3 class

Love this idea. Almost did it myself #1461

Reply:
But @pablo-gar, not that I removed the internal `as_arrow_table()` in #1461 since it was redundant with `soma_array_to_arrow_table()` and conflicted with `arrow::as_arrow_table()`.

Reply:
For correctness, let's add "proposing to remove `as_arrow_table()` in #1461" and maybe we should move a little slower here / not quite do that.

Reply:
> Love this idea. 

Come to think about it, in `tiledb-r` I changed this and am now returning at arrow 'Table` each time.  That is simpler.  Maybe we should do that here too.